.github/workflows/ci.yml

```diff
@@ -89,21 +89,49 @@ jobs:
       - uses: taiki-e/install-action@v2
         with:
           tool: nextest
-      - name: "Test (nextest)"
+      - name: Test (nextest)
         id: nextest
         env:
-          GIX_TEST_IGNORE_ARCHIVES: 1
+          GIX_TEST_IGNORE_ARCHIVES: '1'
         run: cargo nextest --profile=with-xml run --workspace --no-fail-fast
         continue-on-error: true
-      - name: Check how many tests failed
-        if: steps.nextest.outcome == 'failure'
+      - name: Check for errors
+        run: |
+          [xml]$junit_xml = Get-Content -Path 'target/nextest/with-xml/junit.xml'
+          if ($junit_xml.testsuites.errors -ne 0) { exit 1 }
+      - name: Collect actual failures
+        run: |
+          [xml]$junit_xml = Get-Content -Path 'target/nextest/with-xml/junit.xml'
+
+          $actual_failures = $junit_xml.SelectNodes("//testcase[failure]") |
+              ForEach-Object { "$($_.classname) $($_.name)" } |
+              Sort-Object
+
+          Write-Output $actual_failures
+          Set-Content -Path 'actual-failures.txt' -Value $actual_failures
+      - name: Collect expected failures
         env:
-          # See https://github.com/GitoxideLabs/gitoxide/issues/1358.
-          EXPECTED_FAILURE_COUNT: 14
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          $issue = 1358  # https://github.com/GitoxideLabs/gitoxide/issues/1358
+
+          $match_info = gh issue --repo GitoxideLabs/gitoxide view $issue --json body --jq .body |
+              Out-String |
+              Select-String -Pattern '(?s)```text\r?\n(.*?)```'
+
+          $expected_failures = $match_info.Matches.Groups[1].Value -split "`n" |
+              Where-Object { ($_ -match '^\s*FAIL \[') -and ($_ -notmatch '\bperformance\b') } |
+              ForEach-Object { $_ -replace '^\s*FAIL \[\s*\d+\.\d+s\]\s*', '' -replace '\s+$', '' } |
+              Sort-Object
+
+          Write-Output $expected_failures
+          Set-Content -Path 'expected-failures.txt' -Value $expected_failures
+      - name: Compare expected and actual failures
         run: |
-          [xml]$junit = Get-Content -Path 'target/nextest/with-xml/junit.xml'
-          if ($junit.testsuites.errors -ne 0) { exit 1 }
-          if ($junit.testsuites.failures -gt $env:EXPECTED_FAILURE_COUNT) { exit 1 }
+          # Fail on any differences, even unexpectedly passing tests, so they can be investigated.
+          # (If the job is made blocking for PRs, it may make sense to make this less stringent.)
+          git --no-pager diff --no-index --exit-code --unified=1000000 --color=always -- `
```

Comment:
Using git for a general diff between two files is very interesting, I didn't even know that's possible.

Reply:
`diff` in PowerShell is an alias for the `Compare-Object` cmdlet. Of course, the `diff` program on Windows can still be run as `diff.exe`, though I am uncertain if that will find the one installed as part of the Git Bash environment, or another one, when run from PowerShell on GHA runners.

More broadly, over time I've increasingly come to like `git diff`, since it seems to work more similarly across systems than `diff`, which is not a single program but any of a number of implementations. `git diff` is also feature rich and has the benefit of being configurable with Git configuration variables.

This can be made more compact, if desired, by omitting `--exit-code`, which is implied by `--no-index`, omitting `--no-pager` since the inability to page interactively will be detected, by expressing `--color=always` as just `--color`, and possibly even by omitting `--no-index`, since a diff between a tracked and an untracked file works fine without it (though I would want to check if that is still guaranteed to imply `--exit-code`, I am not sure). However, all of these seem to be making clearer how the command is supposed to work.